torch/_functorch/eager_transforms.py

```diff
@@ -338,7 +338,9 @@ def _safe_zero_index(x):
     return x[0]
 
 
-def jacrev(func: Callable, argnums: Union[int, Tuple[int]] = 0, *, has_aux=False):
+def jacrev(func: Callable, argnums: Union[int, Tuple[int]] = 0, *, has_aux=False,
+           chunk_size: Optional[int] = None,
+           _preallocate_and_copy=False):
```

Comment:
cc @samdow @soulitzer @Chillee for API help.

We've got a couple of options here.
1. Either we have a chunk_size argument, or we have a chunks argument (for the number of total chunks).
2. _preallocate_and_copy is private, or we expose it publicly.

My opinion is:
- we should make preallocate_and_copy public. In the long run, memory-planning in PT2 should save us, but idk how soon that is coming and the preallocate_and_copy code is simple enough to maintain.
- I have a slight preference for `chunks`:
    - If jacrev(f)(x) OOMs, the user needs to try out a chunks/chunks_size argument. If the API is `chunks`, then it is clear what the next number the user should try is: 2, and they can keep incrementing this until they're satisfied.  If the API is `chunk_size`, the user just tosses random numbers or needs to compute the size of their jacobian to figure out what the max chunk_size is so they know what the range of numbers to try is.

Reply:
Some nuggets of wisdom from Horace:
1. chunk_size seems nicer, because (1) it is batch-agnostic and (2) it's like loop unrolling - you loop unroll some number of things instead of indicate how many times you want to loop unroll.
2. _preallocate_and_copy should NOT be public. PT2 is around the corner and already includes this optimization (assuming the entire jacrev call can be captured); one of the goals of compilation is to remove code smells like this.

Also, we can always update chunk_size to have different behavior, if we really want.
- if the number is >= 1 then it is a chunk size
- if the number is between 0 and 1 then it is 1/chunks.